### PR TITLE
Circleci docker compose

### PR DESCRIPTION
**Pull request by unenglishable**
* updates dockerignore and gitignore to handle docker.env

* updates Dockerfile to use a specific node base image version (increases build predictability and stability)

* updates cli to include new script for checking database connectivity

* implements docker-compose.yml for running epochtalk with docker compose

* implements circleci-docker-compose.yml for running epochtalk on circleci with docker-compose

* updates circleci.yml to run [api tests](https://github.com/epochtalk/epochtalk-api-tests) with docker-compose

* updates [dockumentation](./DOCKER.md) to explain how to run with docker-compose

### CODE REVIEW

**Comment by unenglishable** [comment]
pending [merge in epochtalk/epoch](https://github.com/epochtalk/epoch/pull/2).

**DO NOT MERGE UNTIL THAT MERGE IS DONE**

File: DOCKER.md [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
-  `echo $(docker-machine ip default):$(docker port epochtalk 8080 | cut -f 2 -d :)`
+### Bring up the EpochTalk service
 
-## Dockumentation
+`docker-compose up`
 
-* start up the thing
+On the first run, this command will download and install containers for
+dependencies, and build the EpochTalk image. Because of this, it will take a bit
+longer to start.  Subsequent startups should be quicker.
```
Comment (akinsey) [inline DOCKER.md]: Broh, be consistent, you have two spaces between sentences here and only one on the line before

Comment (unenglishable) [inline DOCKER.md]: Please end your sentences with a period.

**Comment by unenglishable** [comment]
epoch changes are in master; this pr is good to go.  tested with latest changes in epochtalk/epochtalk master

**Pull request merged**